src/search/pruning_method.h

```diff
@@ -2,29 +2,35 @@
 #define PRUNING_METHOD_H
 
 #include "operator_id.h"
-#include "task_proxy.h"
+
+#include "../utils/timer.h"
 
 #include <memory>
 #include <vector>
 
 class AbstractTask;
+class State;
+
+namespace limited_pruning {
+class LimitedPruning;
+}
 
 class PruningMethod {
+    utils::Timer timer;
+    friend class limited_pruning::LimitedPruning;
+
+    virtual void prune_operators(
+        const State &state, std::vector<OperatorID> &op_ids) = 0;
 protected:
     std::shared_ptr<AbstractTask> task;
-
+    long num_unpruned_successors_generated;
+    long num_pruned_successors_generated;
 public:
     PruningMethod();
     virtual ~PruningMethod() = default;
-
     virtual void initialize(const std::shared_ptr<AbstractTask> &task);
-
-    /* This method must not be called for goal states. This can be checked
-       with assertions in derived classes. */
-    virtual void prune_operators(const State &state,
-                                 std::vector<OperatorID> &op_ids) = 0;
-
-    virtual void print_statistics() const = 0;
+    void prune_op_ids(const State &state, std::vector<OperatorID> &op_ids);
```

Comment:
Hahaha, nice trick to free the name `prune_operators` for the actual internal pruning. But a much uglier name in the public interface. Is there another reason to rename it (e.g. to align it with other names)? Otherwise I would rather keep the public name and find a more precise one for the "internal" method.

Reply:
No, just the reason you found. Any suggestion for the name of the internal method?